docs/source/single_page_run_through.rst

```diff
@@ -0,0 +1,400 @@
+    from fastats.linear_algebra import ols
+
+    def ols_wrap(x):
+        return ols(x[:,:1], x[:,1:])[0][0]
+
+    result = windowed_pass(x, 10, value=ols_wrap)
+
+
+The `ols_wrap` function is due to a limitation of `fastats` - you cannot currently pass a function requiring
+more than one argument to `windowed_pass`; the workaround is to write a one-line wrap function which passes each column
+of the data set correctly to the underlying function, as shown above. This limitation will be removed in a future version.
+
+.. note:: Similar to the `single_pass` function, this `windowed_pass` example will take the high-performance `ols` implementation from the `fastats` standard library, and JIT compile both the OLS and the loop which performs the iteration. In this way the user gets high performance without needing to know the underlying details of performance coding.
+
+This is a quick way to get a rolling OLS function, however usually we are not interested in the slope values - most people
+will be more interested in the `r^2` and `t-statistic` values. To do this we will need to return multiple values from the
+wrap function::
+
+    from fastats import windowed_pass
+    from fastats.linear_algebra import ols, r_squared
+
+    def ols_r_squared(x):
+        out = np.zeros(2)
+        a = x[:, :1]
+        b = x[:, 1:]
+        slope = ols(a, b)[0][0]
+        r2 = r_squared(a, b)[0][0]
+        out[0] = slope
+        out[1] = r2
+        return out
+
+    result = windowed_pass(data, 10, value=ols_r_squared)
+
+.. note:: This wrap function for multiple return values is a bit fiddly to set up, but we are actively working on making this easier.
+
+This works because the return value from the `windowed_pass` function is always the same shape as the input data. As a result
+with the raw input data in two columns, we can support two return values from the inner jitted function. This is however quite
+restrictive, and we will improve this in a future release.
+
+.. automodule:: fastats.core.windowed_pass
+    :members:
+
+windowed_stateful_pass
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. automodule:: fastats.core.windowed_stateful_pass
+    :members:
+
+Early JIT compilation without execution - `return_callable`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Most users will find it useful to store the jitted function, rather than re-jitting the code every time it is called. To
+do this, pass the keyword argument `return_callable=True` to any core function (a `core` function is one with an `@fs` decorator) and
+it will return the function instead of executing it.
+
+For example::
+
+    rolling_ols = windowed_pass(x, 10, value=ols_wrap, return_callable=True)
+
+After this, you can call the `rolling_ols` function without incurring another AST transform and JIT compilation cycle.
+The result is just a normal Python function, so you call it with its expected arguments::
+
+    result = rolling_ols(x, 10)
+    large_window_result = rolling_ols(x, 250)
```

Comment:
Would be nice to include `%timeit` measurements with and without `return_callable=True` to demonstrate the rationale.